src/Docker.DotNet/Endpoints/StreamUtil.cs

```diff
@@ -30,32 +31,47 @@ internal static async Task MonitorStreamAsync(Task<Stream> streamTask, DockerCli
 
         internal static async Task MonitorStreamForMessagesAsync<T>(Task<Stream> streamTask, DockerClient client, CancellationToken cancel, IProgress<T> progress)
         {
-            using (var stream = await streamTask)
+            var serializer = new Newtonsoft.Json.JsonSerializer();
+            try
             {
-                // ReadLineAsync must be cancelled by closing the whole stream.
+                using (var stream = await streamTask)
                 using (cancel.Register(() => stream.Dispose()))
+                using (var reader = new StreamReader(stream, new UTF8Encoding(false)))
+                using (var jsonReader = new JsonTextReader(reader) { SupportMultipleContent = true })
                 {
-                    using (var reader = new StreamReader(stream, new UTF8Encoding(false)))
+                    while (jsonReader.Read())

```

Comment:
If we used ReadAsync on this then we wouldnt need to do the stream.Dispose above on cancellation and then we wouldnt need this complex exception processing anyways right?

Reply:
Haven't found any other way to make events work, events endpoint is failing a for now.
Will try using ReadAsync

Reply:
Could you show me a sample of what you mean? 
This solution is not mine but @simardst
https://github.com/dotnet/Docker.DotNet/issues/434#issue-558268469
I just extended exception catchs as I was running tests

Reply:
So I think that since you switched to the JSON.Net reader we can instead:

```c#
using (var stream = await streamTask)
using (var reader = new StreamReader(stream, new UTF8Encoding(false)))
using (var jsonReader = new JsonTextReader(reader) { SupportMultipleContent = true })
{
    bool done = false;
    do
    {
        done = await jsonReader.ReadAsync(cancel);
        // deserialize, event progress, etc.
    } while (!done)
}
```

Reply:
By doing it this way JSON.Net will return from the reader when cancellation is requested which previously was the issue that "ReadLineAsync" would not because it did not have the cancellation token. We had to close the reader out from under it in this line:
```c#
using (cancel.Register(() => stream.Dispose()))
```
Which we should no longer need

Reply:
@jterry75 It looks like there was a reason behind disposing the stream.